src/main/modules/oidc/index.ts

```diff
@@ -79,6 +79,16 @@ export class OidcMiddleware {
 
       const existingUserCase = await req.locals.api.getExistingUserCase(res.locals.serviceType);
 
+      if (
+        !existingUserCase &&
+        !newInviteUserCase &&
+        config.get('services.case.checkDivCases') &&
+        (await req.locals.api.hasInProgressDivorceCase())
```

Comment:
we should remove the hasInProgressDivorceCase check from the getExistingUserCase function, since we're going to do it separately
also, given that we only care about existing divorce cases when there is no existing nfd case or invite case, we can add it as another else if clause in the findExistingAndNewUserCases function, and set the redirect url there

Reply:
we can, I added this to the top instead because I thought that if/else block is getting a bit ugly and this might be clearer (and clearer for the issue at the moment). If we have time at some point I'd like to try cleaning that block up a bit. I can move it now if you prefer it there though

Reply:
it's not a big deal, just that the hasInProgressDivorceCase is only relevant if there are no existing or invite user cases. so we can save ourselves having to do the check by putting it towards the end

Reply:
ah I see, that makes sense. Will update that - thanks